aws/rust-runtime/aws-config/src/default_provider/app_name.rs

```diff
@@ -117,7 +118,7 @@ mod tests {
     #[tokio::test]
     async fn profile_name_override() {
         let fs = Fs::from_slice(&[("test_config", "[profile custom]\nsdk_ua_app_id = correct")]);
-        let conf = crate::from_env()
+        let conf = crate::from_env_with_version(BehaviorMajorVersion::latest())
```

Comment:
I think we should rename this to `defaults_with_version`, and `load_from_env_with_version` to `load_defaults_with_version`. In fact, maybe even remove the `_with_version` part.